.github/workflows/legend-release.yml

```diff
@@ -26,9 +36,138 @@ env:
   WORKFLOW_ID: legend-stack-release.yml
 
 jobs:
+  compute-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      legend_shared_release_version: ${{ steps.set_legend_shared_release_version.outputs.legend_shared_release_version }}
+      legend_pure_release_version: ${{ steps.set_legend_pure_release_version.outputs.legend_pure_release_version }}
+      legend_engine_release_version: ${{ steps.set_legend_engine_release_version.outputs.legend_engine_release_version }}
+      legend_sdlc_release_version: ${{ steps.set_legend_sdlc_release_version.outputs.legend_sdlc_release_version }}
+      legend_depot_release_version: ${{ steps.set_legend_depot_release_version.outputs.legend_depot_release_version }}
+    steps:
+      - name: Compute legend-shared-release-version
+        id: set_legend_shared_release_version
+        if: github.event.inputs.legend-shared-release != 'n/a'
+        run: |
+          GROUP_ID="org.finos.legend.shared"
+          ARTIFACT_ID="legend-shared"
+          
+          # Query Maven Central API to get latest version
+          latest_version=$(curl -s "https://search.maven.org/solrsearch/select?q=g:${GROUP_ID}+AND+a:${ARTIFACT_ID}&core=gav&rows=1&wt=json" | jq -r '.response.docs[0].v')
+          
+          # Increment patch version
+          IFS='.' read -r major minor patch <<< "$latest_version"
+          
+          release_type="${{ github.event.inputs.legend-shared-release }}"
+          if [ "$release_type" == "major" ]; then
+            major=$((major + 1))
+          elif [ "$release_type" == "minor" ]; then
+            minor=$((minor + 1))
+          else
+            patch=$((patch + 1))
+          fi
+          legend_shared_release_version="${major}.${minor}.${patch}"
+          echo "legend_shared_release_version=${legend_shared_release_version}" >> "$GITHUB_OUTPUT"
+
+      - name: Compute legend-pure-release-version
```

Comment:
This action is getting very long with a lot of duplicate code. Can you move the common elements of this action to dedicated bash files, and then change this action to simply call those bash files and get their output?

Reply:
Done